### ISSUE TITLE
Parsing fails upon reaching a symbolic link to a directory outside obsidian vault

### ISSUE THREAD

**Issue by tprotopopescu** [daeh/zotero-markdb-connect#139]
I have a vault with a symbolic link to a directory outside my obsidian vault. Directories before the link (alphabetically) are searched and markdown files found according to my settings, but directories after the link are ignored. 

The link is called 'Library'; all directories start with A-K are searched, but all directories after 'Library' are ignored. Renaming the link, for example, to start with 'A' causes all folders in the vault to be ignored. 

An obvious workaround is to rename the link starting with 'Zzzz' or something like that - but it seems like symbolic links should be ignored, or there be an option to either ignore them or have the search follow them.

**Comment by daeh** [comment]
What OS are you on? The plugin is currently set up to ignore symbolic links for a few reason but symbolic links shouldn't break the indexing of a folder.

**Comment by tprotopopescu** [comment]
I'm on linux. Opensuse 15.6, kernel 6.4.0-150600.23.25-default, kde 6.2 with wayland. MarkDB 0.1.2, Zotero 7.0.7. 

I just upgraded to Zotero 7, I did not notice this on Zotero 6, though I was not looking so I don't know how much that observation is worth.

**Comment by daeh** [comment]
See if this snippet lists the desired files for you. If works I'll draft a new release with the fix.

`Tools > Developer > Run JavaScript`
Run as async function: `true`
Replace `"PATH/TO/DIR"` with your local path

```js
const listDirContents = async (dirpath) => {
  const items = []
  /* Zotero.File.iterateDirectory calls new OS.File.DirectoryIterator(dirpath) */
  await Zotero.File.iterateDirectory(dirpath, (item) => {
    if (!item.name.startsWith('.')) {
      items.push(item)
    }
  })
  return items
}

const listFilesRecursively = async function* (dirpath) {
  const entries = await listDirContents(dirpath)
  for (const entry of entries) {
    const isSymbolic = Zotero.File.pathToFile(entry.path).isSymlink()
    if (!isSymbolic) {
      if (entry.isDir) {
        yield* listFilesRecursively(entry.path)
      } else {
        yield entry
      }
    }
  }
}

const run = async function (dirpath) {
  const files = []
  for await (const filepath of listFilesRecursively(dirpath)) {
    files.push(filepath)
  }
  return files
}

return run("PATH/TO/DIR")
```

**Comment by daeh** [comment]
Merged in [`v0.1.3`](https://github.com/daeh/zotero-markdb-connect/releases/tag/v0.1.3)

closed by 06fbc92cf1a8237ba1f2075cdf43787d8060abb3
